src/bp-members/bp-members-functions.php

```diff
@@ -1860,19 +1860,15 @@ function bp_core_signup_user( $user_login, $user_password, $user_email, $usermet
 		$activation_key = wp_generate_password( 32, false );
 
 		/**
-		 * WordPress's default behavior is to create user accounts
-		 * immediately at registration time. BuddyPress uses a system
-		 * borrowed from WordPress Multisite, where signups are stored
-		 * separately and accounts are only created at the time of
-		 * activation. For backward compatibility with plugins that may
-		 * be anticipating WP's default behavior, BP silently creates
-		 * accounts for registrations (though it does not use them). If
-		 * you know that you are not running any plugins dependent on
-		 * these pending accounts, you may want to save a little DB
-		 * clutter by defining setting the BP_SIGNUPS_SKIP_USER_CREATION
-		 * to true in your wp-config.php file.
+		 * Filter here to keep creating a user when a registration is performed on regular WordPress configs.
+		 *
+		 * @since 14.0.0
+		 * @todo Fully deprecate in 15.0.0
+		 *
+		 * @param boolean $value True to carry on creating a user when a registration is performed.
+		 *                       False otherwise.
 		 */
-		if ( ! defined( 'BP_SIGNUPS_SKIP_USER_CREATION' ) || ! BP_SIGNUPS_SKIP_USER_CREATION ) {
```

Comment:
For those using this constant, I wonder if we should add a `_doing_it_wrong`. Similar to how we are doing with other similar constants.

Reply:
The default value of this hook could be the value of the `BP_SIGNUPS_SKIP_USER_CREATION` constant itself. In case someone relies on it.